src/parser/grammar.ne

```diff
@@ -210,6 +216,33 @@ between_predicate -> %BETWEEN _ simple_expression _ %AND _ simple_expression {%
   })
 %}
 
+case_expression -> %CASE _ simple_expression:* case_clause:* _ %END {%
```

Comment:
Can you have a CASE with no WHEN , only END ?

Reply:
Probably not, but it's not really important in our case. Just like we allow CASE with only an ELSE or even multiple ELSEs.